src/edg.rs

```diff
@@ -417,7 +567,23 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
                         },
                     );
                 }
-                VertexAssignment::UNDECIDED => {} // fallthrough
+                _ => {
+                    // update distance
+                    let dist = self.distances.get(vertex).unwrap_or(&0);
+                    self.distances
+                        .insert(vertex.clone(), max(dist.clone(), distance));
+
+                    self.mark_interest(vertex, requester);
+
+                    if let Some(dependencies) = self.depends.get(vertex) {
+                        if let Some(assignment) = self.assignment.get(vertex) {
+                            match assignment {
+                                VertexAssignment::UNDECIDED => self.explore(vertex, weight.clone()),
+                                _ => {}
+                            }
```

Comment:
Fixes `clippy` warning
```suggestion
                            if let VertexAssignment::UNDECIDED = assignment {
                                self.explore(vertex, weight.clone());
                            }
```